api-authorization/server/authorization-middleware.js

```diff
@@ -0,0 +1,38 @@
+const jwt = require('jsonwebtoken'); // eslint-disable-line
+const ClientError = require('./client-error'); // eslint-disable-line
+
+function authorizationMiddleware(req, res, next) {
+  /* your code here */
+  const token = req.get('X-Access-Token');
+  try {
+    // eslint-disable-next-line no-unused-expressions
+    token;
+  } catch (err) {
+    throw new ClientError(401, 'authentication required');
+  }
+
+  jwt.verify(token, process.env.TOKEN_SECRET, (error, decoded) => {
```

Comment:
This works, but it's simpler to just use the synchronous version of jwt.verify. If you do not pass a function in the 3rd parameter, then `jwt.verify` simply returns the payload (or throws an error if verification fails).